rdpy/testlib/test_manager.py

```diff
@@ -0,0 +1,61 @@
+from .test_client import RDPTestClientFactory
+import rdpy.core.log as log
+from threading import Event
+
+from twisted.internet import reactor, threads
+
+
+def connect(host, port, username, password, connect_timeout=10.0):
+    """
+    @summary: Connect to a RDP server with the provided details, waits until as
+    session has been established, or times out.
+    @param host: server
+    @param port: port
+    @param username:
+    @param password:
+    @param connect_timeout: Time to wait for the connection.
+    @return: RDPClientTestFactory which holds the connection to the
+    server. use factory.loggedIn to determine if it is connected.
+    """
+    log.info("Connecting to %s@%s:%d" % (username, host, port))
+    started_cond = Event()
+    factory = RDPTestClientFactory(
+        username, password,
+        started_cond)
+    reactor.callFromThread(reactor.connectTCP, host, port, factory)
+    started_cond.wait(connect_timeout)
+    return factory
+
+
+class RDPTestConnManager(object):
+    """
+    @summary: Handles multiple connections to a server, can be used in a with
```

Comment:
In twisted you can make the protocol factor handle tracking multiple connections, thereby negating the need for an extra layer on top. A factory is kind of a "connection manager", so this layer seems to be a connection manager manager! Generally, I'd look after collections of protocols.